File: src/main/java/de/sparkarmy/jda/misc/punishments/PunishmentType.java
package de.sparkarmy.jda.misc.punishments;

public enum PunishmentType {
    UNKNOWN(-1, "unknown"),
    WARN(1, "warn"),
    MUTE(2, "mute"),
    KICK(3, "kick"),
    BAN(4, "ban"),
    TIMEOUT(5, "timeout"),
    UNBAN(6, "unban"),
    SOFTBAN(7, "softban"),

    ;
    private final Integer id;
    private final String name;

    PunishmentType(Integer id, String name) {
        this.id = id;
        this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

}

File: src/main/java/de/sparkarmy/model/PunishmentType.kt
package de.sparkarmy.model

import de.sparkarmy.data.BitField

enum class PunishmentType(override val offset: Int, val identifier: String) : BitField {
    WARN(1, "Warn"),
    MUTE(2, "Mute"),
    KICK(3, "Kick"),
    BAN(4, "Ban"),

    TEMPBAN(14, "Temp-Ban")
}